devito/nodes.py

```diff
@@ -484,31 +495,14 @@ def __repr__(self):
     @property
     def _cparameters(self):
         """Generate arguments signature."""
-        cparameters = []
-        for v in self.parameters:
-            if isinstance(v, Dimension):
-                cparameters.append(v.decl)
-            elif v.is_ScalarFunction:
-                cparameters.append(c.Value('const int', v.name))
-            else:
-                cparameters.append(c.Value(c.dtype_to_ctype(v.dtype),
-                                           '*restrict %s_vec' % v.name))
-        return cparameters
+        return [v.decl for v in self.parameters]
 
     @property
     def _ccasts(self):
         """Generate data casts."""
-        alignment = "__attribute__((aligned(64)))"
-        handle = [f for f in self.parameters
-                  if isinstance(f, (SymbolicData, TensorFunction))]
-        shapes = [(f, ''.join(["[%s]" % ccode(i) for i in f.symbolic_shape[1:]]))
-                  for f in handle]
-        casts = [c.Initializer(c.POD(v.dtype,
-                                     '(*restrict %s)%s %s' % (v.name, shape, alignment)),
-                               '(%s (*)%s) %s' % (c.dtype_to_ctype(v.dtype),
-                                                  shape, '%s_vec' % v.name))
-                 for v, shape in shapes]
-        return casts
+        t_args = [x for x in self.parameters
```

Comment:
calling ``t_args`` something that in the constructor has identical name, but different semantics